corehq/pillows/management/commands/print_elastic_mappings.py

```diff
@@ -0,0 +1,47 @@
+from contextlib import contextmanager
+
+from django.core.management.base import BaseCommand
+
+from corehq.pillows.core import DATE_FORMATS_ARR, DATE_FORMATS_STRING
+from corehq.pillows.mappings.const import NULL_VALUE
+from corehq.pillows.mappings import CANONICAL_NAME_INFO_MAP
+
+from .utils import pprint
+
+MAPPING_SPECIAL_VALUES = {
+    "DATE_FORMATS_ARR": DATE_FORMATS_ARR,
+    "DATE_FORMATS_STRING": DATE_FORMATS_STRING,
+    "NULL_VALUE": NULL_VALUE,
+}
+
+
+class Command(BaseCommand):
+
+    help = "Print Elasticsearch index mappings."
+
+    def add_arguments(self, parser):
+        parser.add_argument("-o", "--outfile", metavar="FILE",
+            help="write output to %(metavar)s rather than STDOUT")
+        parser.add_argument("--no-names", action="store_true", default=False,
+            help="do not replace special values with names")
+        parser.add_argument("cname", metavar="INDEX", choices=sorted(CANONICAL_NAME_INFO_MAP),
+            help="print mapping for %(metavar)s")
+
+    def handle(self, cname, **options):
+        outpath = options["outfile"]
+        if options["no_names"]:
+            namespace = {}
+        else:
+            namespace = MAPPING_SPECIAL_VALUES
+        if outpath is None:
+            outpath = self.stdout._out  # Why, OutputWrapper.write()? Why?
```

Comment:
The `self.stdout` (an instance of `OutputWrapper`) doesn't comply with Duck-typing because its `write()` method automatically writes a trailing newline every time it's called (behaves like `print()`), so doing things like passing it to a JSON encoder has awful results.

Reply:
Does settings `self.stdout.ending = None` not solve the newline problem?

Reply:
Hmm, that would fix the line-ending problem (I didn't know about this `ending` property). I'll also have to contend with the `style_func` function.

I decided to `sys.stdout` instead of `self.stdout`.  This gains me two things:

1. Allows use of `argparse.FileType` for the `--outfile` argument which simplifies the code as @calellowitz recommended.
2. Don't have to muck with Django's `OutputWrapper.ending` and `OutputWrapper.style_func` for guaranteed output consistency.

The [Django docs](https://docs.djangoproject.com/en/2.2/howto/custom-management-commands/) mention that the importance of using `self.stdout` over `sys.stdout` is for testing the output, which is not a concern (`sys.stdout` is simple enough to patch in tests).  Note that `sys.stdout` is necessary with the `FileType` argument type because it explicitly uses `sys.stdout` if the special value `-` is provided as the argument value.